swupd/create_manifests.go

```diff
@@ -257,14 +257,51 @@ func processBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
 
 func addUnchangedManifests(appendTo *Manifest, appendFrom *Manifest) {
 	for _, f := range appendFrom.Files {
-		if f.findFileNameInSlice(appendTo.Files) == nil {
-			if f.Name == IndexBundle {
-				// this is generated new each time
-				continue
-			}
-			appendTo.Files = append(appendTo.Files, f)
+		if f.Name == IndexBundle {
+			// this is generated new each time
+			continue
+		}
+		if f.Type == TypeManifest &&
+			f.findFileNameInSlice(appendTo.Files) != nil {
+			continue
+		}
+		if f.Type == TypeIManifest &&
+			f.findIManifestInSlice(appendTo.Files) != nil {
```

Comment:
If you do what I suggest above you don't have to use two different implementations to find the manifest in appendTo

Reply:
I see you have to do some special work to determine an iterative manifest name. Two implementations may be the best way to handle that.

Reply:
I think I'd prefer adding the argument like you mention just to simplify the logic in this function.